### PR TITLE
cpu/cc2538: add Watchdog implementation

### PR DESCRIPTION
**Pull request by benpicco**
The Watchdog on the CC2538 only supports 4 intervals (2ms, 16ms, 250ms & 1s).
Since the watchdog timer API specifies a `max_time`, the interval equal or
below that time is selected.

E.g. for `max_time=125ms` the 16ms interval would be selected.
This is outside the tolerance of the `tests/periph_wdt` test.

### Testing procedure

`tests/periph_wdt` will fail as the watchdog internals are too coarse for the test. You will see e.g.

```
FAILED target time: 128[ms], actual_time: 13.98[ms]
```

250 ms would be > `max_time`, so the next interval (16ms) is selected.

### Issues/PRs references

taken over from #7374

### CODE REVIEW

**Comment by fjmolinas** [comment]
```
2020-08-31 09:19:33,145 # RIOT wdt test application
setup 0 1000
2020-08-31 09:19:52,215 #  setup 0 1000
2020-08-31 09:19:52,215 # valid configuration
> start
2020-08-31 09:19:53,879 #  start
2020-08-31 09:19:53,880 # starting wdt timer
> 2020-08-31 09:19:54,903 #  main(): This is RIOT! (Version: 2020.10-devel-1113-g18fe16-pr-14890)
2020-08-31 09:19:54,905 # RIOT wdt test application
```

Tested works fine.

**Comment by fjmolinas** [comment]
@benpicco what about applying this patch to the test script:

```diff
 # directory for more details.
 
 import sys
+import os
 import pexpect
 import time
+import subprocess
 from testrunner import run
 
+# nucleo-l152re is the default platform for this test
+BOARD = os.environ.get("BOARD", "nucleo-l152re")
+CPU = subprocess.check_output([
+    'make',
+    'info-debug-variable-CPU',
+    'BOARD={}'.format(BOARD),
+    '--no-print-directory']).strip()
+
 # We test only up to 10ms, with smaller times mcu doesn't have time to
 # print system time before resetting
-reset_times_ms = [128, 512, 1024, 8192]
+# cc2538 only supports 4 intervals [2ms, 16ms, 250ms, 1s]
+if CPU.decode("utf-8")  == 'cc2538':
+    reset_times_ms = [16, 250, 1000]
+else:
+    reset_times_ms = [128, 512, 1024, 8192]
 
 # We don't check for accuracy, only order of magnitude. Some MCU use an
 # an internal un-calibrated clock as reference which can deviate in
```

**Review by fjmolinas — APPROVED** [review]
Looks good, tested and it works.

File: tests/periph_wdt/tests/01-run.py [hunk 1]
```diff
@@ -7,13 +7,27 @@
...
+import subprocess
 from testrunner import run
 
+# nucleo-l152re is the default platform for this test
+BOARD = os.environ.get("BOARD", "nucleo-l152re")
+CPU = subprocess.check_output([
+    'make',
+    'info-debug-variable-CPU',
+    'BOARD={}'.format(BOARD),
+    '--no-print-directory']).strip()
```
Comment (benpicco) [inline tests/periph_wdt/tests/01-run.py:22]: looks like Travis doesn't like this :confused:

Comment (fjmolinas) [reply tests/periph_wdt/tests/01-run.py:22]: Lets just change the variable name, it does not need to me `CPU`, can be `BOARD_CPU` and I think travis would not complain.

**Pull request merged**